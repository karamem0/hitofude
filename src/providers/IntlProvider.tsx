//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import { IntlProvider as Provider, createIntl } from 'react-intl';
import en from '../translations/en.json';
import ja from '../translations/ja.json';

const translations: { [key: string]: Record<string, string> } = {
  en,
  ja
};

const intl = createIntl({
  defaultLocale: 'en',
  locale: window.navigator.language.substring(0, 2),
  messages: translations[window.navigator.language]
});

function IntlProvider(props: Readonly<React.PropsWithChildren<unknown>>) {

  const { children } = props;

  return (
    <Provider {...intl}>
      {children}
    </Provider>
  );

}

export default IntlProvider;
