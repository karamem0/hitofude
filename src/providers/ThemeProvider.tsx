//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import { FluentProvider as Provider, Theme } from '@fluentui/react-components';
import { InvalidOperationError } from '../types/Error';
import { ThemeName } from '../types/Model';
import { themes } from '../themes/Theme';

interface ThemeContextState {
  theme: Theme,
  themeName: ThemeName,
  changeTheme: (themeName: ThemeName) => void
}

const ThemeContext = React.createContext<ThemeContextState | undefined>(undefined);

export const useTheme = (): ThemeContextState => {
  const value = React.useContext(ThemeContext);
  if (value == null) {
    throw new InvalidOperationError();
  }
  return value;
};

function ThemeProvider(props: Readonly<React.PropsWithChildren<unknown>>) {

  const { children } = props;

  const [ themeName, setThemeName ] = React.useState<ThemeName>(ThemeName.light);

  const handleChangeTheme = React.useMemo(() => (value: ThemeName) => {
    setThemeName(value);
  }, []);

  const value = React.useMemo<ThemeContextState>(() => ({
    theme: themes[themeName],
    themeName,
    changeTheme: handleChangeTheme
  }), [
    themeName,
    handleChangeTheme
  ]);

  return (
    <ThemeContext.Provider value={value}>
      <Provider theme={value.theme}>
        {children}
      </Provider>
    </ThemeContext.Provider>
  );

}

export default ThemeProvider;
