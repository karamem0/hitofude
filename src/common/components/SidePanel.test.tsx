//
// Copyright (c) 2023-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import IntlProvider from '../../providers/IntlProvider';
import Presenter from './SidePanel.presenter';
import ThemeProvider from '../../providers/ThemeProvider';

it('should create a shapshot when the open parameter is true', () => {
  const container = document.body.appendChild(document.createElement('div'));
  const params = {
    children: (
      <div data-testid="test-Children" />
    ),
    mountNode: container,
    open: true,
    title: 'Side Panel'
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>,
    {
      container
    }
  );
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByTestId('test-Children')).toBeInTheDocument();
});

it('should create a shapshot when the open parameter is false', () => {
  const container = document.body.appendChild(document.createElement('div'));
  const params = {
    children: (
      <div data-testid="test-Children" />
    ),
    mountNode: container,
    open: false,
    title: 'Side Panel'
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>,
    {
      container
    }
  );
  expect(asFragment()).toMatchSnapshot();
  expect(screen.queryByTestId('test-Children')).not.toBeInTheDocument();
});
