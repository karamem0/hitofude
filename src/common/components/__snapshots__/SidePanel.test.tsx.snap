// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should create a shapshot when the open parameter is false 1`] = `
<DocumentFragment>
  <div
    class="fui-FluentProvider fui-FluentProviderr3 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  />
</DocumentFragment>
`;

exports[`should create a shapshot when the open parameter is true 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="rmnljus fui-OverlayDrawer__backdrop ___mgf8q40_1ccxybz f1euv43f"
  />
  .emotion-0 {
  height: 100svh;
}

.emotion-1 {
  margin-bottom: 1rem;
}

<aside
    aria-labelledby="dialog-title-r2"
    aria-modal="true"
    class="fui-OverlayDrawer r7dfyj8 emotion-0 ___xtj47k0_1v5b7dm fsqykmd f1e31b4d frppm18 fqofjzu f1euv43f"
    data-tabster="{"restorer":{"type":0},"modalizer":{"id":"modal-r1","isOthersAccessible":false,"isTrapped":true}}"
    role="dialog"
    tabindex="-1"
  >
    <header
      class="fui-DrawerHeader r3ovn4i"
      role="none"
    >
      <div
        class="fui-DrawerHeaderTitle ___1g6dz7n_17w6zju f22iagw f1869bpl f122n59 fsnqrgy"
      >
        <h2
          class="fui-DrawerHeaderTitle__heading fui-DialogTitle rxjm636"
          id="dialog-title-r2"
        >
          Side Panel
        </h2>
        <div
          class="fui-DrawerHeaderTitle__action fui-DialogTitle__action r13kcrze ___zdrx2v0_7c5v900 faqnl2i"
        >
          <button
            aria-label="Close"
            class="fui-Button r1alrhcs ___w3o4yv0_df8ghi0 fhovq9v f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb f1t94bn6 f1s2uweq fr80ssc f1ukrpxl fecsdlb fnwyq0v ft1hn21 fuxngvv fy5bs14 fsv2rcd f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1omzyqd f1dfjoow f1j98vj9 fj8yq94 f4xjyn1 f1et0tmh f9ddjv3 f1wi8ngl f18ktai2 fwbmr0d f44c6la"
            type="button"
          >
            <span
              class="fui-Button__icon rywnvv2"
            >
              <svg
                aria-hidden="true"
                class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
                fill="currentColor"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m2.59 2.72.06-.07a.5.5 0 0 1 .63-.06l.07.06L8 7.29l4.65-4.64a.5.5 0 0 1 .7.7L8.71 8l4.64 4.65c.18.17.2.44.06.63l-.06.07a.5.5 0 0 1-.63.06l-.07-.06L8 8.71l-4.65 4.64a.5.5 0 0 1-.7-.7L7.29 8 2.65 3.35a.5.5 0 0 1-.06-.63l.06-.07-.06.07Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </header>
    <div
      class="fui-DrawerBody ri6rvx7 emotion-1"
      role="group"
      tabindex="0"
    >
      <div
        data-testid="test-Children"
      />
    </div>
  </aside>
  <div
    class="fui-FluentProvider fui-FluentProviderr0 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <span
      hidden=""
    />
  </div>
</DocumentFragment>
`;
